### DISCUSSION TITLE
PK-Parameter in Time Profile Analysis - multiple questions

### DISCUSSION THREAD

**Discussion by PavelBal** [Open-Systems-Pharmacology/Forum#1065] [Questions & Problems]
While going through the documentation (https://docs.open-systems-pharmacology.org/working-with-pk-sim/pk-sim-documentation/pk-sim-simulations#pk-analysis-view) and comparing it with the actual output of the pk-parameters in PK-Sim, I realized I have multiple questions.

Following use case: I have a model with two compounds - Esomeprazole and R-Omeprazole (these are the two enantiomeres of Omeprazole). A container observer for the molecule Esomeprazole calculates the concentration of the sum of Esomeprazole and R-Omeprazole in Pervipheral Venous Blood Plasma. This observer is called "Omeprazole Plasma (Peripheral Venous Blood)"

In an iv simulation (both enantiomeres are administered via iv route), I select only the observer "Omeprazole Plasma (Peripheral Venous Blood)" for the output, but not the concentrations of the single compounds. When I open the PK-Analysis, I see this:

![image](https://user-images.githubusercontent.com/25903040/146546909-2cef7908-5976-4303-b579-92585eee15fa.png)

1) For the general understanding: the PK-parameters in the lower table are calculated for the selected outputs (and observed data, if available), and the PK-parameters in the upper table are calculated for all molecules in the model, even if I do not select their outputs?

2) How are the "_norm"-parameters (e.g. AUC_tEnd_norm) calculated in this case? For the observer which sums up concentrations of two compounds, I assume it only accounts for the dose of the molecule for which the observer is calculated (Esomeprazole in this case). If so, the calculated value is not correct. In fact, allowing user-defined observers introduces a problem here.

3) Why is "AUC Ratio (AUCR)" calculated for the compounds only, and not for the observer? Here actually it could be calculated (as opposed to 2) ). Same for C_max ratio.

4) I cannot find parameter "Total body clearance".

5) Instead, I see parameter  "Total body clearance/F", which I cannot find in the documentation at all.

6) Also parameters "Vd(plasma)/F" and "Vss(plasma)/F", which should not appear for the iv simulation.

@msevestre @Yuri05 @StephanSchaller

**Comment by PavelBal** [comment]
Now, a po simulation. Same setting - both enantiomeres are administered, and the observer for the total concentration is selected as output.

![image](https://user-images.githubusercontent.com/25903040/146549962-681da415-6cea-43a7-a744-e99ef7ae6104.png)

7) How are the parameters "Vss(plasma)/F", "Vd(plasma)/F", "Total plasma clearance/F", and "Total Body Clearance/F" calculated? I did not press "Calculate Bioavailability" so far, so which values does the calculation use?

8) The parameter "Total body clearance/F" cannot be found in the documentation.

9) After calculating bioavailability, I see this:

![image](https://user-images.githubusercontent.com/25903040/146550669-63062128-73ab-46bc-be3d-b83fa3a98e30.png)


Could it be, that the "/F" and not "/F" captions are mixed-up?

10) Why, after calculation bioavailability, the non- /F parameters (which are labeled as /F but probably wrong...) are not shown any more?

**Comment by PavelBal** [comment]
Complex administration protocol: a compound is administered as iv at time 0, and as po at time 120h.

11) Why cannot the "Fraction absorbed" be calculated?
12) Same question for "Vss (plasma)", "Vd (plasma)", and "Total plasma clearance"
13) AUC_tEnd, AUC_inf, %AUC(tlast-inf), AUC_tEnd_norm, AUC_inf_norm, C_tEnd are not calculated (though stated in documenation as "for all simulations")

**Comment by msevestre** [comment]
Some quick feedback:
The definition of the parameter/F (for instance Total plasma clearance/F) does not require F. The way it is calculated is based on the IV définition. 
CL/F = Dose/AUC

If you want to know the real value, you need to multiply by F
That's why Total plasma clearance is only available once you have the bioavailability 
Value of parameter P = P/F x F 
So there is no inversion of parameters whatsoever

**Comment by msevestre** [comment]
>For the general understanding: the PK-parameters in the lower table are calculated for the selected outputs (and observed data, if available), and the PK-parameters in the upper table are calculated for all molecules in the model, even if I do not select their outputs?

This is correct. Some outputs are ALWAYS used, even if not selected. Those are 
https://github.com/Open-Systems-Pharmacology/PK-Sim/blob/develop/src/PKSim.Core/Services/SimulationPersistableUpdater.cs#L35

* Venous Blood Plasma
* Peripheral Venous Blood Plasma
* F abs oral 

>How are the "_norm"-parameters (e.g. AUC_tEnd_norm) calculated in this case? For the observer which sums up concentrations of two compounds, I assume it only accounts for the dose of the molecule for which the observer is calculated (Esomeprazole in this case). If so, the calculated value is not correct. In fact, allowing user-defined observers introduces a problem here.

I think this is a bug. I need to think about it

>Why is "AUC Ratio (AUCR)" calculated for the compounds only, and not for the observer? Here actually it could be calculated (as opposed to 2) ). Same for C_max ratio.

This is calculated for all compounds (and only once using peripheral venous blood except for mouse and rat where it is using venous blood). This is how it was specified. We have 100s of observers so I am not quite sure what you mean with calculated for observer

>I cannot find parameter "Total body clearance".

This should be Total body clearance/F

>Instead, I see parameter "Total body clearance/F", which I cannot find in the documentation at all. 

See above

>Also parameters "Vd(plasma)/F" and "Vss(plasma)/F", which should not appear for the iv simulation.

I don't think those parameter makes sense for oral simulation either... (at least they make sense in the global view but not for each selected output)..

**Comment by msevestre** [comment]
>Why, after calculation bioavailability, the non- /F parameters (which are labeled as /F but probably wrong...) are not shown any more?

The rest if I have explained. Here I guess we could show the /F parameters as well. not sure why we are hiding them

**Comment by msevestre** [comment]
moving this to the forum as discussion so that it can be found later on

**Comment by msevestre** [comment]
>Why cannot the "Fraction absorbed" be calculated?

It can be calculated but I am not sure if this is meaningful. Let say you have oral and iv and iv starts before the oral application ends etc.. I am not sure that we would be able to differentiate the part that comes from the oral application and the one coming from the iv.  At any rate, in code, it is explicitly shown for oral application(s) only. Maybe @Yuri05  can comment on this
https://github.com/Open-Systems-Pharmacology/PK-Sim/blob/develop/src/PKSim.Core/Services/GlobalPKAnalysisTask.cs#L117

> Same question for "Vss (plasma)", "Vd (plasma)", and "Total plasma clearance"

Those parameters are only define for single dosing

>AUC_tEnd, AUC_inf, %AUC(tlast-inf), AUC_tEnd_norm, AUC_inf_norm, C_tEnd are not calculated (though stated in documenation as "for all simulations")

They are also only defined for single dosing applications

**Comment by msevestre** [comment]
FYI
![image](https://user-images.githubusercontent.com/1041237/147596711-11bcf55c-e722-45d9-97f2-500850b25104.png)

**Comment by PavelBal** [comment]
> This is calculated for all compounds (and only once using peripheral venous blood except for mouse and rat where it is using venous blood). This is how it was specified. We have 100s of observers so I am not quite sure what you mean with calculated for observer

I mean the observers that are selected by the user. I might be interested in the AUC ratio in muscle intracellular (why not?).

> It can be calculated but I am not sure if this is meaningful. Let say you have oral and iv and iv starts before the oral application ends etc.. I am not sure that we would be able to differentiate the part that comes from the oral application and the one coming from the iv. At any rate, in code, it is explicitly shown for oral application(s) only. Maybe @Yuri05 can comment on this

The observer for fraction absorbed uses the parameters `MOLECULE|Total oral drug mass` and `Neighborhoods|Lumen_sto_Lumen_duo|MOLECULE|Oral mass absorbed`, so I do not see how the amount administered via iv route would interfere here...

> Those parameters are only define for single dosing

Yes, but why? :D

C_tEnd is just the concentration at the end of simulation... It should not matter how many applications are simulated? The same for AUC - I may be very much interested what the AUC in a multiple dose simulation is.

**Comment by PavelBal** [comment]
> Those parameters are only define for single dosing

lol, even more confusing! I have a simulation with two IV (0 and 24h) and and PO (129h) administrations. Simulation time is 140h:

![image](https://user-images.githubusercontent.com/25903040/148088926-05780360-d239-4771-a091-011d30e594e5.png)

I get the long list for of PK-parameters for simulations with multiple administrations... Now, change the simulation time to the time of administration of the last dose (129h):

I get the PK parameters for the single dose! i.e., the **are** calculated even for multiple dosing!

![image](https://user-images.githubusercontent.com/25903040/148089670-1880e243-fefa-4583-94f4-566846bf9155.png)

**Reply by msevestre** [reply]
Could you share that project? I am curious to understand why. That looks like a bug to me but I want to make sure you did not change something else

**Reply by StephanSchaller** [reply]
I can confirm and get the same behaviour (single dose PK Property list if SimTime of a multiple dose scenario ends with a dosing time and full list if not) for a different project also for population simulations

**Comment by PavelBal** [comment]
[Fluvoxamine-Omeprazole-DDI_light.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/7809060/Fluvoxamine-Omeprazole-DDI_light.zip)

Just change the simulation time to 139h

**Comment by StephanSchaller** [comment]
Another point I found: the 'C_trough_tDlast' actually calculates the 'C_end' and not the 'C' BEFORE the last dose. 
E.g., if I finish simulating just 1h after the last dose it takes the last concentration value:
![image](https://user-images.githubusercontent.com/25477643/148096303-14077907-ef20-43f1-9894-d5513ffc073d.png)

**Reply by msevestre** [reply]
yes see : https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/932
